test/cpp/api/module.cpp

```diff
@@ -297,6 +293,44 @@ TEST_F(ModuleTest, CloneCreatesDistinctParameters) {
   }
 }
 
+TEST_F(ModuleTest, CloneCreatesDistinctParameters) {
+  auto module = std::make_shared<TestDistinctParametersModule>();
+  torch::NoGradGuard no_grad;
+  auto module2 = module->clone();
+  testDistinctParameters(module, module2);
+}
+
+TEST_F(ModuleTest, CloneCreatesDistinctParametersExplicitDevice_CUDA) {
+  auto module = std::make_shared<TestDistinctParametersModule>();
+  torch::NoGradGuard no_grad;
+  torch::Device device(torch::kCUDA, 0);
+  module->to(device);
+  auto module2 = module->clone(device);
+  testDistinctParameters(module, module2);
+}
+
+TEST_F(ModuleTest, CloneCreatesDistinctParametersExplicitDevice_MultiCUDA) {
```

Comment:
This test doesn't seem to run in `pytorch_linux_xenial_cuda9_cudnn7_py3_multigpu_test` (https://circleci.com/gh/pytorch/pytorch/2244426?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link). It might be worthwhile to fix it before landing this PR, to make sure that we are running multi-gpu tests for libtorch.

Reply:
Thanks, I agree. Just haven't got a chance to work on that yet, and will fix that before landing this PR.